mlflow/utils/file_utils.py

```diff
@@ -230,6 +227,12 @@ def from_json(input_var):
             return json.load(f)
 
     j2_env.filters["from_json"] = from_json
+    # Compute final source of context file (e.g. my-profile.yml), applying Jinja filters
+    # like from_json as needed to load context information from files, then load into a dict
+    context_source = j2_env.get_template(context_name).render({})
+    context_dict = yaml.safe_load(context_source)
```

Comment:
Is it acceptable to use `safe_load` to load the final context file contents (`profile.yml`) from a string? Should we be passing a custom `Loader` here or something? Looks like `safe_load` doesn't accept any parameters besides an input string - so if we do want to use our custom `Loader` we could write the final `profile.yml` to a tempfile and load it from there

Reply:
Shall we still apply UniqueKeyLoader? Looks like that's missing here.

Reply:
Ah, yeah no need to actually use `safe_load` here, using `yaml.load()` and passing `UniqueKeyLoader` works perfectly fine - thanks @dbczumar !